doctr/models/modules/vision_transformer/tensorflow.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2022, Mindee.
+
+# This program is licensed under the Apache License 2.0.
+# See LICENSE or go to <https://opensource.org/licenses/Apache-2.0> for full license details.
+
+from typing import Any, Tuple
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+from doctr.utils.repr import NestedObject
+
+__all__ = ["PatchEmbedding"]
+
+
+class PatchEmbedding(layers.Layer, NestedObject):
+    """Compute 2D patch embeddings with cls token and positional encoding"""
+
+    def __init__(
+        self,
+        img_size: Tuple[int, int],
+        patch_size: Tuple[int, int],
+        embed_dim: int,
+    ) -> None:
```

Comment:
`PatchEmbedding` signature in TensorFlow and PyTorch are not identical: there is `channels` kwarg in PyTorch for Conv2D but it's not needed for TensorFlow.

To have a common signature, maybe you can pass the whole `input_shape` (i.e. Image Size + Channels) in both TensorFlow and PyTorch implementation, and add `input_shape` as kwarg in `layers.Conv2D` in Tensorflow.

I must admit it's not needed, it's more a "cosmetic" suggestion.

Reply:
yep lets change it :)